storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java

```diff
@@ -262,6 +291,77 @@ private Database createV6Database() {
     }
   }
 
+  /**
+   * LevelDb database uses same setup as v4 but backed by leveldb instead of RocksDb
+   *
+   * @return the created database
+   */
+  private Database createLevelDbV1Database() {
+    try {
+      final V5DatabaseMetadata metaData =
+          V5DatabaseMetadata.init(getMetadataFile(), V5DatabaseMetadata.v5Defaults());
+      DatabaseNetwork.init(getNetworkFile(), Constants.GENESIS_FORK_VERSION, eth1Address);
+      return RocksDbDatabase.createLevelDb(
+          metricsSystem,
+          metaData.getHotDbConfiguration().withDatabaseDir(dbDirectory.toPath()),
+          metaData.getArchiveDbConfiguration().withDatabaseDir(v5ArchiveDirectory.toPath()),
+          stateStorageMode,
+          stateStorageFrequency,
+          specProvider);
+    } catch (final IOException e) {
+      throw DatabaseStorageException.unrecoverable("Failed to read metadata", e);
+    }
+  }
+
+  private Database createLevelDbV2Database() {
+    try {
+      final V6DatabaseMetadata defaultMetaData;
+      if (v6ArchiveDirectory.isPresent()) {
+        defaultMetaData = V6DatabaseMetadata.separateDBDefault();
+      } else {
+        defaultMetaData = V6DatabaseMetadata.singleDBDefault();
+      }
+
+      final V6DatabaseMetadata metaData =
+          V6DatabaseMetadata.init(getMetadataFile(), defaultMetaData);
+      if (defaultMetaData.isSingleDB() != metaData.isSingleDB()) {
+        throw DatabaseStorageException.unrecoverable(
+            "The database was originally created as "
+                + (metaData.isSingleDB() ? "Single" : "Separate")
+                + " but now accessed as "
+                + (defaultMetaData.isSingleDB() ? "Single" : "Separate"));
+      }
+
+      DatabaseNetwork.init(getNetworkFile(), Constants.GENESIS_FORK_VERSION, eth1Address);
+
+      final RocksDbConfiguration hotOrSingleDBConfiguration =
+          metaData.isSingleDB()
+              ? metaData.getSingleDbConfiguration().orElseThrow().getConfiguration()
+              : metaData.getSeparateDbConfiguration().orElseThrow().getHotDbConfiguration();
+
+      final Optional<RocksDbConfiguration> finalizedConfiguration =
+          v6ArchiveDirectory.map(
+              dir ->
+                  metaData
+                      .getSeparateDbConfiguration()
+                      .orElseThrow()
+                      .getArchiveDbConfiguration()
+                      .withDatabaseDir(dir.toPath()));
+
+      return RocksDbDatabase.createLevelDbV2(
```

Comment:
Here or in a follow-up may be worth renaming `RocksDbDatabase` etc to something like `KeyValueDatabase` or `ColumnarDatabase`.

Reply:
Yep, planning that for a follow up if we stick with LevelDB.  Probably need to rearrange some packages as well.

Reply:
Logged https://github.com/ConsenSys/teku/issues/3588